gpt_mini/modeling/tensorflow_char_gpt.py

```diff
@@ -254,7 +232,8 @@ def _respond(
         return context
 
     def train(self) -> None:
-        text_dict = self._get_data()
+        datalayer = DataLayer(data_source=self._data_source)
```

Comment:
I still think the data should not be read within the train method/function for the sole reason that it makes things lot to debug.